src/lib/crawler/crawler.ts

```diff
@@ -1,68 +1,222 @@
 import axios from "axios";
 import * as cheerio from "cheerio";
+import Bottleneck from "bottleneck";
 
-export interface MatchDetails {
+// provided by initial GET call w/ character name & realm
+export interface MatchSummary {
+  matchId: string;
   team_name: string;
-  date: string;
   bracket: string;
-  arena: string;
-  points_change: string;
-  character_details: CharacterDetail[]; // | undefined[]; I don't think it can be undefined
-  id: string;
   outcome: string;
+  points_change: string;
+  date: string;
   duration: string;
-  team?: string;
+  arena: string;
 }
 
+// provided by secondary POST call w/ match ID
 export interface CharacterDetail {
-  matchmaking_change?: string;
-  healingDone: string;
+  realm: string;
+  charname: string;
+  class?: string;
   race?: string;
   gender?: string;
-  killingBlows: string;
+  teamname: string;
   teamnamerich: string;
-  realm: string;
   damageDone: string;
-  charname: string;
-  class?: string;
   deaths: string;
+  healingDone: string;
+  killingBlows: string;
+  matchmaking_change?: string;
   personal_change: string;
-  teamname: string;
+}
+
+// MatchSummary & CharacterDetail merged
```

Comment:
In my opinion, merged doesn't accurately describe this interface. Merging one interface with another sounds more like what happens when you `extend`

```TypeScript
interface MatchDetails extends CharacterDetail
```

Also, I haven't really done a style review yet, and it should really be more on me to set the style standards, but I think you should consider the usefulness of some of these comments. I can easily read the interface and see that an array CharacterDetail is a property of MatchDetails. See my suggestion for a comment that explains something that may not be so obvious.

```suggestion
/**
 * MatchDetails is an aggregate of all data that the crawler collects 
 * for a given matchId. These objects are intended to be stored
 * long term in some kind of database for further use and analysis
 */
```
Writing good comments and documentation is relatively difficult. In your studying, you might have seen tips where people aggressively encourage new developers to write comments, but a useless comment is worse than no comment in my opinion. My recommendation is try to think of the type of people who will be reading this code in the future including yourself.